packages/@dcl/sdk-commands/src/logic/project-files.ts

```diff
@@ -74,3 +75,81 @@ export async function getProjectContentMappings(
 
   return ret
 }
+
+export interface IFile {
+  path: string
+  content: Buffer
+  size: number
+}
+
+export const MAX_FILE_SIZE_BYTES = 50 * 1e6 // 50mb
+
+/**
+ * Returns a promise of an array containing all the file paths for the given directory.
+ * @param dir The given directory where to list the file paths.
+ */
+export const getAllFilePaths = async (
+  components: Pick<CliComponents, 'fs'>,
+  dir: string,
+  ignoredFiles: Ignore,
+  relativeDir: string = dir
+): Promise<string[]> => {
+  try {
+    const files = await components.fs.readdir(dir)
+    let tmpFiles: string[] = []
+
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i]
+      const filePath = resolve(dir, file)
+      const relativePath = relative(relativeDir, filePath)
+      const stat = await components.fs.stat(filePath)
+
+      if (!ignoredFiles.test(relativePath).ignored) {
+        if (stat.isDirectory()) {
+          const folderFiles = await getAllFilePaths(components, filePath, ignoredFiles, relativePath)
+          tmpFiles = tmpFiles.concat(folderFiles)
+        } else {
+          tmpFiles.push(filePath)
+        }
+      }
+    }
+
+    return tmpFiles
+  } catch (e) {
+    return []
+  }
+}
+
+/**
+ * Returns a promise of an array of objects containing the path and the content for all the files in the project.
+ * All the paths added to the `.dclignore` file will be excluded from the results.
+ * Windows directory separators are replaced for POSIX separators.
+ * @param ignoreFile The contents of the .dclignore file
+ */
+export const getFiles = async (
```

Comment:
please use `getPublishableFiles().filter(size)` for this validation

Reply:
the thing is, just a `filter` won't do it since we also need to map the result into `{ path: string, content: buffer, size: number }`. We could `reduce` them into `Promise.all` but in my opinion, that's uglier than using a for loop